conan/tools/gnu/autotoolsdeps.py

```diff
@@ -0,0 +1,62 @@
+from conan.tools.env import Environment
+
+
+class AutotoolsDeps(object):
+    def __init__(self, conanfile):
+        # Set the generic objects before mapping to env vars to let the user
+        # alter some value
+        self._conanfile = conanfile
+        deps_cpp_info = conanfile.deps_cpp_info
+        self.libs = list(deps_cpp_info.libs)
+        self.libs.extend(list(deps_cpp_info.system_libs))
+        self.include_paths = list(deps_cpp_info.include_paths)
+        self.library_paths = list(deps_cpp_info.lib_paths)
+        self.defines = list(deps_cpp_info.defines)
+        self.cflags = list(deps_cpp_info.cflags)
+        self.cxx_flags = list(deps_cpp_info.cxxflags)
+        self.sharedlinkflags = list(deps_cpp_info.sharedlinkflags)
+        self.exelinkflags = list(deps_cpp_info.exelinkflags)
+        self.frameworks = list(deps_cpp_info.frameworks)
+        self.frameworks_paths = list(deps_cpp_info.framework_paths)
+        self.sysroot = deps_cpp_info.sysroot
+
+    def generate(self):
+        # cpp_flags
+        cpp_flags = []
+        include_paths = ['-I"%s"' % p for p in self.include_paths]
+        cpp_flags.extend(include_paths)
+        cpp_flags.extend(["-D%s" % define for define in self.defines])
+
+        # Libs
+        libs = ["-l%s" % library for library in self.libs]
+
+        # Ldflags
+        # TODO: Discuss, should the helper filter frameworks based on compiler?
```

Comment:
I vote to keep the least possible logic in these helpers. if some filtering has to be done, it doesn't belong here. and IMO it's the responsibility of the recipe author to ensure the correct CppInfo is built for the given configuration.